kernel/src/main/java/org/kframework/kil/Module.java

```diff
@@ -57,5 +57,19 @@ public void toString(StringBuilder sb) {
         sb.append("endmodule");
     }
 
-
+    /**
+     * Used to determine if a module was modified.
+     * <a href="https://github.com/runtimeverification/k/issues/2910"> GH #2910</a>
+     * This allows for scripts to move files around after compilation but still give error messages
+     * if the contents of modules differ.
+     */
+    public long digest() {
+        StringBuilder mod = new StringBuilder();
+        toString(mod);
+        long hash = 7;
+        for (int i = 0; i < mod.length(); i++) {
```

Comment:
this is not at all a collision resistant hash function and is thus not suitable as a proxy for equality.